subsys/net/ip/net_context.c

```diff
@@ -3181,6 +3181,51 @@ int net_context_get_option(struct net_context *context,
 	return ret;
 }
 
+int net_context_get_local_addr(struct net_context *ctx,
+			       struct sockaddr *addr,
+			       socklen_t *addrlen)
+{
+	if (ctx == NULL || addr == NULL || addrlen == NULL) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_TCP) &&
+	    net_context_get_type(ctx) == SOCK_STREAM) {
+		return net_tcp_endpoint_copy(ctx, addr, NULL, addrlen);
```

Comment:
> We usually cannot use context->local for established TCP connections because the local address is not updated for TCP.

Wouldn't that indicate some sort of bug in our network stack? I think one could expect to be able to access `context->local` address reliably, regardless of the net_context type.

Reply:
I can investigate this more, it is just that if the application does not bind to some specific address, the `context->local` is set to `::`. We do not update the value in this case when the connection is established, don't exactly remember the reasoning for that, but faintly recall there was some issues in it. This part of the code is quite complex and not easy to see all the ramifications.